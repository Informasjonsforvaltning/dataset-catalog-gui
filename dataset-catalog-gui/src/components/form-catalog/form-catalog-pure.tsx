import React, { memo, FC, useState } from 'react';
import { compose } from 'redux';
import { Field } from 'redux-form';
import cx from 'classnames';
import { Collapse } from 'reactstrap';

import {
  withTranslations,
  Props as TranslationsProps
} from '../../providers/translations';

import Translation from '../translation';
import Helptext from '../helptext/helptext.component';
import InputTitleField from '../fields/field-input-title/field-input-title.component';
import TextAreaField from '../fields/field-textarea/field-textarea.component';
import { PencilInlineIcon } from '../../fdk-icons/icons';

interface ExternalProps {
  initialValues?: any;
  values?: any;
  isReadOnly?: boolean;
}

interface Props extends ExternalProps, TranslationsProps {}

const FormCatalog: FC<Props> = ({
  initialValues: { title, publisher },
  values,
  isReadOnly,
  translationsService
}) => {
  const [collapseTitle, setCollapseTitle] = useState(false);
  const [collapse, setCollapse] = useState(false);

  const toggleTitle = () => setCollapseTitle(!collapseTitle);

  const toggleDescription = () => setCollapse(!collapse);

  const collapseClass = cx(
    'fdk-reg_collapse',
    'fdk-reg_backgroundDefault',
    'fdk-datasets-description',
    {
      'fdk-reg_collapse_open': collapse
    }
  );

  const fieldClass = cx('fdk-title-input', {
    'w-100': collapseTitle
  });

  return (
    <form className='mb-5 fdk-reg-catalogs'>
      <div className='d-flex align-items-center justify-content-between'>
        {title && title.nb && !collapseTitle && (
          <h1 className='w-75 fdk-text-strong'>
            <Translation object={title} />
          </h1>
        )}
        {!isReadOnly && (
          <div className={fieldClass}>
            <Field
              name={`title.${translationsService.getLanguage()}`}
              component={InputTitleField}
              hideInput={collapseTitle}
              onToggleTitle={toggleTitle}
            />
          </div>
        )}
      </div>

      {publisher && publisher.name && (
        <div className='fdk-reg-datasets-publisher mt-2 mb-4'>
          <Translation id='schema.catalog.ownedByLabel' /> {publisher.name}
        </div>
      )}

      <div className={collapseClass}>
        <div className='d-flex justify-content-between w-100'>
          <div className='d-flex fdk-color-neutral-darkest'>
            <Translation object={values.description} />
          </div>
          {!isReadOnly && (
            <button
              type='button'
              onClick={e => {
                e.preventDefault();
                toggleDescription();
              }}
            >
              <PencilInlineIcon />
              <Translation id='schema.catalog.editDescriptionLabel' />
            </button>
          )}
        </div>
        {!isReadOnly && (
          <Collapse className='mt-3' isOpen={collapse}>
            <div className='form-group'>
              <Helptext
                title={translationsService.translate(
                  'schema.catalog.helptext.title'
                )}
                required
                term='Catalog_title'
              />
              <Field
                name={`description.${translationsService.getLanguage()}`}
                component={TextAreaField}
                label={translationsService.translate(
                  'schema.common.descriptionLabel'
                )}
              />
            </div>
          </Collapse>
        )}
      </div>
    </form>
  );
};

export default compose<FC<ExternalProps>>(memo, withTranslations)(FormCatalog);
