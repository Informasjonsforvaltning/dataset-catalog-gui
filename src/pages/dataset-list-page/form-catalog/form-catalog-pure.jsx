import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import cx from 'classnames';
import { Collapse } from 'reactstrap';

import localization from '../../../services/localization';
import { getTranslateText } from '../../../services/translateText';
import { Helptext } from '../../../components/helptext/helptext.component';
import InputTitleField from '../../../components/fields/field-input-title/field-input-title.component';
import TextAreaField from '../../../components/fields/field-textarea/field-textarea.component';

export class FormCatalogPure extends React.Component {
  constructor(props) {
    super(props);
    this.toggleTitle = this.toggleTitle.bind(this);
    this.toggleDescription = this.toggleDescription.bind(this);
    this.state = {
      collapseTitle: false,
      collapse: false
    };
  }

  toggleTitle() {
    const { collapseTitle } = this.state;
    this.setState({ collapseTitle: !collapseTitle });
  }

  toggleDescription() {
    const { collapse } = this.state;
    this.setState({ collapse: !collapse });
  }

  render() {
    const { initialValues, values } = this.props;
    const { collapse, collapseTitle } = this.state;
    const { title, publisher } = initialValues;

    const collapseClass = cx(
      'fdk-reg_collapse',
      'fdk-reg_backgroundDefault',
      'fdk-datasets-description',
      {
        'fdk-reg_collapse_open': collapse
      }
    );

    const fieldClass = cx('fdk-title-input', {
      'w-100': collapseTitle
    });

    return (
      <form className="mb-5 fdk-reg-catalogs">
        <div className="d-flex align-items-center justify-content-between">
          {title && title.nb && !collapseTitle && (
            <h1 className="w-75 fdk-text-strong">
              {getTranslateText(title, localization.getLanguage())}
            </h1>
          )}
          <div className={fieldClass}>
            <Field
              name={`title.${localization.getLanguage()}`}
              component={InputTitleField}
              hideInput={collapseTitle}
              onToggleTitle={this.toggleTitle}
            />
          </div>
        </div>

        {publisher && publisher.name && (
          <div className="fdk-reg-datasets-publisher mt-2 mb-4">
            {localization.schema.catalog.ownedByLabel} {publisher.name}
          </div>
        )}

        <div className={collapseClass}>
          <div className="d-flex justify-content-between w-100">
            <div className="d-flex fdk-color-neutral-darkest">
              {getTranslateText(values.description, localization.getLanguage())}
            </div>
            <button
              type="button"
              onClick={e => {
                e.preventDefault();
                this.toggleDescription();
              }}
            >
              <i className="fa fa-pencil mr-2" />
              {localization.schema.catalog.editDescriptionLabel}
            </button>
          </div>
          <Collapse className="mt-3" isOpen={collapse}>
            <div className="form-group">
              <Helptext
                title={localization.schema.catalog.helptext.title}
                required
                term="Catalog_title"
              />
              <Field
                name={`description.${localization.getLanguage()}`}
                component={TextAreaField}
                label={localization.schema.common.descriptionLabel}
              />
            </div>
          </Collapse>
        </div>
      </form>
    );
  }
}

FormCatalogPure.defaultProps = {
  initialValues: null,
  values: null
};
FormCatalogPure.propTypes = {
  initialValues: PropTypes.object,
  values: PropTypes.object
};
