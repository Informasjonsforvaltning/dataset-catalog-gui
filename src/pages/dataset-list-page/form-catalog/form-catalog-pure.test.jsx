import React from 'react';
import { shallow } from 'enzyme';
import { FormCatalogPure } from './form-catalog-pure';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    initialValues: {
      title: 'Title',
      publisher: 'Publisher'
    },
    values: {
      description: {
        nb: 'Beskrivelse'
      }
    }
  };
  wrapper = shallow(<FormCatalogPure {...defaultProps} />);
});

test('should render FormCatalogPure correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
