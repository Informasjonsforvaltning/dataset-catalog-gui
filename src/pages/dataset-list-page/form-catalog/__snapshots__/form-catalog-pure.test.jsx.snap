// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormCatalogPure correctly 1`] = `
<form
  className="mb-5 fdk-reg-catalogs"
>
  <div
    className="d-flex align-items-center justify-content-between"
  >
    <div
      className="fdk-title-input"
    >
      <Field
        component={[Function]}
        hideInput={false}
        name="title.nb"
        onToggleTitle={[Function]}
      />
    </div>
  </div>
  <div
    className="fdk-reg_collapse fdk-reg_backgroundDefault fdk-datasets-description"
  >
    <div
      className="d-flex justify-content-between w-100"
    >
      <div
        className="d-flex fdk-color-neutral-darkest"
      >
        Beskrivelse
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-pencil mr-2"
        />
        Rediger beskrivelse
      </button>
    </div>
    <Collapse
      appear={false}
      className="mt-3"
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <div
        className="form-group"
      >
        <withStateHandlers(Component)
          required={true}
          term="Catalog_title"
          title="Beskrivelse"
        />
        <Field
          component={[Function]}
          label="Beskrivelse"
          name="description.nb"
        />
      </div>
    </Collapse>
  </div>
</form>
`;
